apps/avatax/src/modules/saleor/shipping-line.ts

```diff
@@ -1,39 +1,90 @@
-import { LineItemModel } from "avatax/lib/models/LineItemModel";
+import { Result, err, ok } from "neverthrow";
+import { z } from "zod";
+import { BaseError } from "../../error";
 import { SHIPPING_ITEM_CODE } from "../avatax/calculate-taxes/avatax-shipping-line";
 
-export class SaleorShippingLine {
-  private getAmount({
-    gross,
-    net,
-    taxIncluded,
-  }: {
-    gross: number;
-    net: number;
-    taxIncluded: boolean;
-  }) {
-    return taxIncluded ? gross : net;
+export interface ISaleorOrderConfirmedShippingLine {
+  getAmount(): number;
+  getIsTaxIncluded(): boolean;
+  getIsDiscounted(): boolean;
+  getQuantity(): number;
+  getItemCode(): string;
+  getShippingTaxCode(): string | undefined;
+}
+
+export class SaleorOrderConfirmedShippingLine implements ISaleorOrderConfirmedShippingLine {
+  public static schema = z.object({
+    gross: z.object({
+      amount: z.number(),
+    }),
+    net: z.object({
+      amount: z.number(),
+    }),
+  });
+
+  private constructor(
+    private data: z.infer<typeof SaleorOrderConfirmedShippingLine.schema>,
+    // order config
+    private isTaxIncluded: boolean,
+    private isDiscounted: boolean,
+    // avatax config
+    private shippingTaxCode: string | undefined,
+  ) {}
+
+  public static ParsingError = BaseError.subclass("SaleorOrderConfirmedShippingLineParsingError");
+
+  public static create(
+    payload: unknown,
+    isTaxIncluded: boolean,
+    isDiscounted: boolean,
+    shippingTaxCode: string | undefined,
```

Comment:
in constructur its fine-ish but in public methods we should avoid so many arguments, especially where you have 

```
 isTaxIncluded: boolean,
    isDiscounted: boolean,
```

its few sips of coffee that will make me to swap these fields accidentally